### PR TITLE
Configure mdc auth properties from cookie auth

### PR DESCRIPTION
**Pull request by robert3005**

### CODE REVIEW

File: auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java [hunk 1]
```diff
@@ -41,17 +43,34 @@
     public static final String SESSION_ID_KEY = "sessionId";
     public static final String TOKEN_ID_KEY = "tokenId";
 
+    private final String cookieAuthKey;
+
+    public BearerTokenLoggingFilter(String cookieAuthKey) {
+        this.cookieAuthKey = cookieAuthKey;
```
Comment (carterkozak) [inline auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java]: this is a break, should be optional/default with a no-arg ctor.

File: auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java [hunk 2]
```diff
@@ -41,17 +43,34 @@
     public static final String SESSION_ID_KEY = "sessionId";
     public static final String TOKEN_ID_KEY = "tokenId";
 
+    private final String cookieAuthKey;
```
Comment (carterkozak) [inline auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java]: Conjure allows endpoints to specify different cookie names, how do we plan to register a separate filter instance for each endpoint?

**Review by carterkozak — CHANGES_REQUESTED** [review]

File: auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java (hunk 1 above)
Comment (robert3005) [reply auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java]: ok

File: auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java (hunk 2 above)
Comment (robert3005) [reply auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java]: I believe we fall on convention here - just like I could define authorization to be under different auth headers. Unless you want to get in the game of generating this logging handler for every endpoint separately but right now our server infra doesn't allow that in all cases and you have already handled the other case

Comment (carterkozak) [reply auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java]: >  just like I could define authorization to be under different auth headers

Conjure only allows the `Authorization` to be used for header auth.

> Unless you want to get in the game of generating this logging handler for every endpoint separately

I already am, our undertow handlers have already solved this problem  ;-) 

I think we can solve this for jersey using a DynamicFeature to check the first arg for an auth cookie, without requiring per-app customization or additional code generation.

Comment (robert3005) [reply auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BearerTokenLoggingFilter.java]: Sure, while I agree with you I don't think logging is covered under spec in any form. Once it added we can adjust it. You'll notice that it's only our server implementation uses this filter and not the conjure-java-runtime part. 

Don't get me wrong - I think what you're proposing is probably the right thing to do just don't think it's valuable right now

**Review by iamdanfox — APPROVED** [review]
I did a code-search internally and it seems that all `@CookieParam` usages internally use the same value, so perhaps it's OK to just hard-code that in WC?

**Comment by carterkozak** [comment]
@iamdanfox Did your search include conjure definitions using auth:header as well?

**Comment by iamdanfox** [comment]
Yeah - interestingly I actually didn't find usages of authType: "cookie" in conjure definitions (all the `authType: "header"` ones are covered by our existing BearerTokenLoggingFilter).

**Comment by iamdanfox** [comment]
OK I've put up a quick PR to explore the idea of using a DynamicFeature: https://github.com/palantir/auth-tokens/pull/90.

**Comment by robert3005** [comment]
gonna close in favour of #90

**Pull request closed without merging**